### PR TITLE
Add option `ana.thread.include-node` to disable nodes in TIDs

### LINKED ISSUE #527: Switch to not include creation node / location in thread id
For the incremental analysis it is problematic if thread ids contain the location of the `pthread_create(...)` call as that leads to a complete reanalysis of the threads created in a changed function, even if these threads themselves are unchanged.

It should be configurable via a flag whether the location becomes part of the thread id or not.

### PR DESCRIPTION
**Pull request by michael-schwarz**
For the incremental analysis it is problematic if thread ids contain the node of the pthread_create(...) call as that leads to a complete reanalysis of the threads created in a changed function, even if these threads themselves are unchanged.

Add an option `ana.thread.include-node` to toggle this behavior.

Disabling it does (together with our must joined analysis) sometimes lead to less precision (not distinguishing threads started in sequence that start with the same function ). Interestingly, it can also lead to better precision when the same thread is created in different branches, as the variable holding the thread ids contains only this single unique thread.

See examples `36/97` and `36/98` for examples of this.

Closes #527

### CODE REVIEW

File: src/util/options.schema.json [hunk 1]
```diff
@@ -783,6 +783,13 @@
               "type": "string",
               "enum": ["history", "plain"],
               "default": "history"
+            },
+            "include-loc" : {
+              "title": "ana.thread.include-loc",
+              "description":
+                "Whether the line at which a thread is created is part of its threadid",
```
[the remark this answers is not in the document]
Comment (michael-schwarz) [reply src/util/options.schema.json:790]: `ana.thread.include-node` then?

[the remark this answers is not in the document]
Comment (sim642) [reply src/util/options.schema.json:790]: Or maybe just `ana.thread.node`?

I think there are other references to location still, like the module being called `FunLoc`.

[the remark this answers is not in the document]
Comment (michael-schwarz) [reply src/util/options.schema.json:790]: I think `ana.thread.include-node` is a bit clearer than just `node`....

**Review by sim642 — APPROVED** [review]
Disabling this brings the knot_comb negation patch reanalysis from ~8s to ~2s.

I wonder if I should debug that further...

**Comment by michael-schwarz** [comment]
> Disabling this brings the knot_comb negation patch reanalysis from ~8s to ~2s.

Maybe try with `dbg.print_tids` (on the full run) if the output there is as expected?

**Comment by sim642** [comment]
> Maybe try with `dbg.print_tids` (on the full run) if the output there is as expected?

That is the same for the from-scratch and incremental runs, so this problem is fixed:
```
Encountered number of thread IDs (unique): 9 (5)
unique:  ([thread_process_client, thread_main_autospawn, main], {})  ([thread_process_client, thread_main, main], {})  ([thread_main, main], {})  ([main], {})  ([thread_main_autospawn, main], {}) 
non-unique:  ([thread_main, main], {thread_process_client})  ([main], {thread_main})  ([main], {thread_process_client, thread_main})  ([thread_main_autospawn, main], {thread_process_client}) 
```

**Pull request merged**